app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmissionController.scala

```diff
@@ -43,16 +43,20 @@ class RegistrationSubmissionController @Inject() (
     registrationRepository.get(id).flatMap {
       case Some(registration) =>
         registrationValidationService.validateRegistration(registration) match {
-          case Valid(eclSubscription) =>
-            subscriptionService.subscribeToEcl(eclSubscription, registration).map { response =>
-              nrsService.submitToNrs(
-                registration.base64EncodedNrsSubmissionHtml,
-                response.success.eclReference
-              )
+          case Valid(either) =>
+            either match {
+              case Left(eclSubscription) =>
+                subscriptionService.subscribeToEcl(eclSubscription, registration).map { response =>
+                  nrsService.submitToNrs(
+                    registration.base64EncodedNrsSubmissionHtml,
+                    response.success.eclReference
+                  )
 
-              Ok(Json.toJson(response.success))
+                  Ok(Json.toJson(response.success))
+                }
+              case _                     => Future.successful(BadRequest)
```

Comment:
Why `BadRequest` ?

Reply:
Also missing a unit tests for this `Right(_)` case in `RegistrationSubmissionControllerSpec`

Reply:
Changed to indicate that work still needs to be done.